### PR TITLE
Detailed Wallet info view

### PR DESCRIPTION
**Pull request by tynes**
Wallet Info component to display in depth info about a wallet

- [x] Basic layout
- [x] Rebase onto `BoxGrid`
- [x] User can create accounts
- [x] List transaction history

Still needs final styling, but the functionality is present
Definitely suspect this view will struggle with purse scale

### CODE REVIEW

**Comment by tynes** [comment]
screen shot 2018-10-22 at 2 50 24 pm

**Comment by tynes** [comment]
@bucko13 ready for review

File: lib/mappings/walletInfo.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  selectWalletInfo,
+  selectMultisigWalletInfo,
+  selectSelectedWallet,
+  selectSelectedAccount,
+  selectWalletAccounts,
+  selectWalletAccountInfo,
+  selectHistory,
+} from './selectors';
+
+// mapped to constants/WALLET_INFO_ACCOUNTS_LIST_HEADERS
```
Comment (tynes) [inline lib/mappings/walletInfo.js]: `bwallet/constants/...`?

Comment (bucko13) [reply lib/mappings/walletInfo.js]: what is mapped to the constants here? Not following the comment.

**Comment by bucko13** [comment]
Also getting a plugin crash when I try and click on add account:

```

**Comment by tynes** [comment]
The problem you are seeing is due to an out of date `bpanel-ui` plugin. The dummy Link update hasn't been applied

File: lib/containers/MainContainer.js [hunk 2]
```diff
@@ -30,10 +31,12 @@ class MainContainer extends PureComponent {
     return {};
   }
 
+  // TODO: bug, on refresh, adds all wallets to wallets
+  // even if they are multisig wallets
   async componentDidMount() {
     const { getWallets, getWalletsInfo } = this.props;
 
-    await getWallets();
+    await getWallets('all');
```
[the remark this answers is not in the document]
Comment (tynes) [reply lib/containers/MainContainer.js]: It does default to `all`, was purposefully being explicit - https://github.com/bpanel-org/bwallet/blob/master/lib/actions/wallet.js#L101

[the remark this answers is not in the document]
Comment (bucko13) [reply lib/containers/MainContainer.js]: Gotcha! Didn't see that in the switch statement. I personally like having it in the signature as it feels more explicit, but that's just a personal preference.

File: lib/containers/MainContainer.js [hunk 3]
```diff
@@ -73,12 +76,13 @@ class MainContainer extends PureComponent {
...
-    const { walletSideNavMetadata, addSideNav } = this.props;
-
-    for (const metadata of walletSideNavMetadata) addSideNav(metadata);
+    const { walletsNavMeta, addSideNav } = this.props;
+    for (const metadata of walletsNavMeta) addSideNav(metadata);
   }
 
   render() {
+    // assuming that match.url is /bwallet
+    // changing the metadata path will break things
```
[the remark this answers is not in the document]
Comment (tynes) [reply lib/containers/MainContainer.js]: Exactly, I tried to update it to make it make more sense

File: lib/containers/MainContainer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tynes) [reply lib/containers/MainContainer.js]: Having a default value in the signature you mean?

[the remark this answers is not in the document]
Comment (bucko13) [reply lib/containers/MainContainer.js]: yeah, like `function getWallets(type='all') {` on https://github.com/bpanel-org/bwallet/blob/master/lib/actions/wallet.js#L92

**Comment by bucko13** [comment]
So do I need to link to an unpublished bpanel-ui update or should this be out with the last release? Maybe just need to update the version in package.json

**Comment by tynes** [comment]
Its in the latest published package: https://github.com/bpanel-org/bpanel-ui/blob/v0.0.13/src/components/Link/Link.js

**Comment by bucko13** [comment]
Will probably need to update the package.json and the package-lock then in this PR to the latest version `0.0.13` so that gets installed.

File: lib/containers/MainContainer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tynes) [reply lib/containers/MainContainer.js]: Actually I misread the code, it defaults to `standard` or just normal wallets. There are parts of the codebase that rely on this, I can update it if you have strong preference for it to default to `all`

[the remark this answers is not in the document]
Comment (bucko13) [reply lib/containers/MainContainer.js]: maybe put that in the signature then? I don't have any preference though, whatever you think it easiest to work with. Just a comment from reading through the code.

**Comment by tynes** [comment]
Bumped `bpanel-ui` version

File: lib/containers/MainContainer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tynes) [reply lib/containers/MainContainer.js]: I don't think it should default to using something that not all chains will have, just updated the fn signature

File: lib/mappings/walletInfo.js (hunk 1 above)
Comment (tynes) [reply lib/mappings/walletInfo.js]: Let me know if the new comment doesn't clear this up

File: lib/containers/WalletInfo.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+  update(key, value) {
+    this.setState({ [key]: value });
+  }
+
+  async createAccount() {
+    const { createAccount, selectedWallet } = this.props;
+    const { accountId, passphrase } = this.state;
+
+    // TODO: better way to alert user
```
[the remark this answers is not in the document]
Comment (tynes) [reply lib/containers/WalletInfo.js]: Created a task in asana for this

**Pull request merged**